projects/dsp-ui/src/lib/viewer/services/value-type.service.ts

```diff
@@ -78,6 +79,56 @@ export class ValueTypeService {
                 objectType === valueType;
     }
 
+    /**
+     * Given a date, checks if its precision is supported by the datepicker.
+     *
+     * @param date date to be checked.
+     */
+    private checkPrecision(date: KnoraDate): boolean {
+        return date.precision === Precision.dayPrecision;
+    }
+
+    /**
+     * Given a date, checks if its era is supported by the datepicker.
+     *
+     * @param date date to be checked.
+     */
+    private checkEra(date: KnoraDate): boolean {
+        return date.era === 'CE' || date.era === 'AD';
+    }
+
+    /**
+     * Determines if a date or period can be edited using this component.
+     *
+     * @param date the date or period to be edited.
+     */
+    isDateEditable(date: KnoraDate | KnoraPeriod): boolean {
+
+        // only day precision is supported by the MatDatepicker
+        let precisionSupported: boolean;
+        // only common era is supported by the MatDatepicker
+        let eraSupported: boolean;
+
+        if (date instanceof KnoraDate) {
+            precisionSupported = this.checkPrecision(date);
+            eraSupported = this.checkEra(date);
+        } else {
+            precisionSupported = this.checkPrecision(date.start) && this.checkPrecision(date.end);
+            eraSupported = this.checkEra(date.start) && this.checkEra(date.end);
+        }
+
+        return precisionSupported && eraSupported;
+    }
+
+    /**
+     * Determines if a text can be edited using the text editor.
+     *
+     * @param textValue the text value to be checked.
+     */
+    isTextEditable(textValue: ReadTextValueAsXml): boolean {
+        return textValue.mapping === 'http://rdfh.ch/standoff/mappings/StandardMapping';
+    }
+
     /**
      * Equality checks with constants below are TEMPORARY until component is implemented.
```

Comment:
@mdelez Maybe this comment should be adapted since there are also other reasons to make a value read-only. What do you think?

Reply:
Maybe this service can also be renamed? It's becoming more general and is now more of a place to add helpful functions for the value components